crates/ark/tests/environment.rs

```diff
@@ -279,3 +289,234 @@ fn test_environment_list() {
+    let evt: VariablesFrontendEvent = serde_json::from_value(data).unwrap();
+    match evt {
+        VariablesFrontendEvent::Refresh(params) => {
+            assert_eq!(params.variables.len(), 1);
+            assert_eq!(params.variables[0].display_name, ".Last.value");
+            assert_eq!(params.version, 1);
+        },
+        _ => panic!("Expected refresh event"),
+    }
+
+    // Create a variable in the R environment
+    r_task(|| unsafe {
+        let test_env = test_env.get().clone();
+        let sym = r_symbol!("test_var");
+        Rf_defineVar(sym, Rf_ScalarInteger(99), *test_env);
+    });
+
+    // Simulate a prompt signal
+    EVENTS.console_prompt.emit(());
+
+    // Wait for the update event
+    let msg = outgoing_rx.recv().unwrap();
+    let data = match msg {
+        CommMsg::Data(data) => data,
+        _ => panic!("Expected data message"),
+    };
+
+    // We might get multiple update events - first for .Last.value, then for test_var
+    // Rather than assert on specific quantities, just verify that eventually
+    // both .Last.value and test_var appear
+
+    // Create sets to accumulate variables we've seen
+    let mut seen_last_value = false;
+    let mut seen_test_var = false;
+
+    // Process first update
+    let evt: VariablesFrontendEvent = serde_json::from_value(data).unwrap();
+    match evt {
+        VariablesFrontendEvent::Update(params) => {
+            println!("Update params: {:?}", params);
+
+            // Update what we've seen
+            seen_last_value = params
+                .assigned
+                .iter()
+                .any(|v| v.display_name == ".Last.value") || seen_last_value;
+
+            seen_test_var = params
+                .assigned
+                .iter()
+                .any(|v| v.display_name == "test_var") || seen_test_var;
+        },
+        _ => panic!("Expected update event"),
+    }
+
+    // If we haven't seen both variables yet, try to get a second update
+    if !seen_last_value || !seen_test_var {
+        // It's possible that we won't get another update, so use a timeout
+        if let Ok(msg) = outgoing_rx.recv_timeout(std::time::Duration::from_millis(500)) {
+            if let CommMsg::Data(data) = msg {
+                let evt: VariablesFrontendEvent = serde_json::from_value(data).unwrap();
+                if let VariablesFrontendEvent::Update(params) = evt {
+                    println!("Second update params: {:?}", params);
+
+                    // Update what we've seen
+                    seen_last_value = params
+                        .assigned
+                        .iter()
+                        .any(|v| v.display_name == ".Last.value") || seen_last_value;
+
+                    seen_test_var = params
+                        .assigned
+                        .iter()
+                        .any(|v| v.display_name == "test_var") || seen_test_var;
+                }
+            }
+        }
+    }
+
```

Comment:
oh I guess it would be hard to test the _contents_ of `.Last.value` here? We'd need a full integration test that actually instantiates a full kernel to which we could send `execute_request`, probably not worth it.

Reply:
Yeah, this would require a lot of overhead.